compiler/rustc_type_ir/src/solve/mod.rs

```diff
@@ -185,6 +185,9 @@ pub enum BuiltinImplSource {
     /// This can be removed when `feature(tuple_unsizing)` is stabilized, since we only
     /// use it to detect when unsizing tuples in hir typeck.
     TupleUnsizing,
+    /// A built-in impl that is considered trivial, without any nested requirements. They
+    /// are preferred over where-clauses, and we want to track them explicitly.
+    Trivial,
```

Comment:
vibe: pls move them to the top of this enum

Reply:
👍